code/hash/Hacl.Hash.Core.SHA2.fst

```diff
@@ -340,10 +338,10 @@ let shuffle a block hash ws =
 inline_for_extraction
 let zero (a: sha2_alg): word a =
   match a with
-  | SHA2_224 | SHA2_256 -> 0ul
-  | SHA2_384 | SHA2_512 -> 0UL
+  | SHA2_224 | SHA2_256 -> u32 0
+  | SHA2_384 | SHA2_512 -> u64 0
 
-#set-options "--max_fuel 0 --max_ifuel 0"
+#set-options "--z3rlimit 200 --max_fuel 2 --max_ifuel 2"
```

Comment:
Same as above, copied from _dev but the increase is unneeded here. I set fuel and ifuel to 0. It might become necessary when switching to _dev's precomputed ws though.